### PR TITLE
feat(dispatch): real M10 SLA colour + parcel ref on DA tasks; station…

### PR DESCRIPTION
**Pull request by BanerjeeAgniva**
image

image

- Task RAG now reflects the authoritative M10 per-shipment SLA state (via a new common ShipmentSlaPort, implemented in sla over SlaShipmentRepository) instead of dispatch's local expectedEta heuristic; the heuristic stays as a fallback for tasks with no SLA row yet (pre-pickup). DaTaskItem gains shipmentRef (via the existing ShipmentRefPort) for a parcel-detail link plus actByAt/urgencyMinutes.
- New station-scoped GET /dispatch/da/{daId}/trail returns the DA's GPS breadcrumb for a date, reusing the daDetail city guard (da_gps_ping has no city column) and delegating to the existing DaStatusService.listTrack. Distinct path from the DA-self /track.

Tests: SLA colour overrides the ETA heuristic; trail city guard + delegation.

# Pull Request

## Summary

---

## What Changed

- 
- 

---

## Why This Change?

---

## Testing
- [ ] Tested locally
- [ ] Edge cases considered
- [ ] No breaking changes introduced

### Test Evidence

---

### Areas Affected

- [ ] Backend
- [ ] Frontend
- [ ] Routing
- [ ] Infra
- [ ] Database
- [ ] NA

### Modules Affected

- [ ] M1 — Auth
- [ ] M2 — Pricing
- [ ] M3 — Grid
- [ ] M4 — Orders
- [ ] M5 — Dispatch
- [ ] M6 — Routing
- [ ] M7 — Hub
- [ ] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [ ] NA

---

## Risks / Concerns

- 

---

## Checklist
- [ ] Code follows project conventions (package layout, no cross-module internal imports)
- [ ] Self-review completed
- [ ] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [ ] No sensitive data or credentials exposed
- [ ] Documentation updated if behaviour changed
- [ ] Ready for review

## New Features
* Station managers can view a delivery associate’s GPS breadcrumb trail for a selected date, subject to access permissions.
* Delivery task details now include shipment references, lifecycle timestamps, SLA deadlines, breach status, and urgency information.
* Added delivery associate scorecards with pace, success rate, on-time rate, and activity-based ordering.

## Bug Fixes
* Improved city-based access controls for delivery associate details and GPS trails.

## Tests
* Added coverage for scorecard calculations, ordering, date boundaries, and SLA-related task insights.

### CODE REVIEW

File: dispatch/src/test/java/com/oneday/dispatch/service/impl/DispatchMetricsServiceImplTest.java [hunk 1]
```diff
@@ -30,7 +31,14 @@ class DispatchMetricsServiceImplTest {
...
             mock(com.oneday.common.port.DaDirectoryPort.class);
-    private final DispatchMetricsServiceImpl svc = new DispatchMetricsServiceImpl(repo, directory);
+    private final com.oneday.common.port.ShipmentSlaPort slaPort =
+            mock(com.oneday.common.port.ShipmentSlaPort.class);
+    private final com.oneday.common.port.ShipmentRefPort refPort =
+            mock(com.oneday.common.port.ShipmentRefPort.class);
+    private final com.oneday.dispatch.service.DaStatusService daStatus =
+            mock(com.oneday.dispatch.service.DaStatusService.class);
+    private final DispatchMetricsServiceImpl svc =
+            new DispatchMetricsServiceImpl(repo, directory, slaPort, refPort, daStatus);
```
[the remark this answers is not in the document]
Comment (BanerjeeAgniva) [reply dispatch/src/test/java/com/oneday/dispatch/service/impl/DispatchMetricsServiceImplTest.java:41]: Not applicable — Mockito's default answer returns an empty Map (ReturnsEmptyValues) for Map-returning methods, not null, so slaFor(...)/refsFor(...) yield Map.of() when unstubbed and daDetail doesn't NPE. Verified: DispatchMetricsServiceImplTest runs 8/8 green on this branch, including daDetailSortsByUrgencyAndAttachesIdentity and historyIsSevenDaysZeroFilledAndBounded. No change made.
